src/Hive/Controllers/UserController.cs

```diff
@@ -50,7 +62,9 @@ public class UserController : ControllerBase
         private readonly PermissionsManager<PermissionContext> permissions;
 
         private const string RenameActionName = "hive.user.rename";
-        private PermissionActionParseState renameParseState;
+        private const string GetUserInfoActionName = "hive.user.info";
+        [ThreadStatic] private PermissionActionParseState renameParseState;
+        [ThreadStatic] private PermissionActionParseState getUserParseState;
```

Comment:
Does the compiler really not yell at you for this? `[ThreadStatic]` is meant to be used on `static` fields. I don't know what it would even do on instance fields.